doc/sphinx/proof-engine/tactics.rst

```diff
@@ -650,180 +649,339 @@ Applying theorems
+      By default no renaming is done.
 
-      This also provides apply with values for instantiating premises. Here, variables
-      are referred by names and non-dependent products by increasing numbers (see
-      :ref:`bindings`).
+      .. deprecated:: 8.15
 
-      .. flag:: Apply With Renaming
+         Provided for compatibility with versions prior to 8.15.
 
-         When on, this flag causes the names in the :n:`@term`'s type to be renamed for uniqueness.
-         By default no renaming is done.
+   .. _apply_backward:
+   .. example:: Backward reasoning in the goal with `apply`
 
-         .. deprecated:: 8.15
+      .. coqtop:: reset none
+
+         Goal forall A B C: Prop, (A -> B -> C) -> C.
+
+      .. coqtop:: out
+
+         intros A B C H.
+
+      .. coqtop:: all
 
-            This flag is provided for compatibility with versions prior to 8.15.
+         apply H.  (* replace goal with new goals for unmatched premises of H *)
 
-   .. tacv:: apply {+, @term}
+   .. _apply_forward:
+   .. example:: Forward reasoning in hypotheses with `apply`
 
-      This is a shortcut for :n:`apply @term__1; [.. | ... ; [ .. | apply @term__n] ... ]`,
-      i.e. for the successive applications of :n:`@term`:sub:`i+1` on the last subgoal
-      generated by :n:`apply @term__i` , starting from the application of :n:`@term__1`.
+      .. coqtop:: reset none
+
+         Goal forall A B C: Prop, B -> (A -> B -> C) -> True.
+
+      .. coqtop:: out
+
+         intros A B C H0 H1.
+
+      .. coqtop:: all
 
-   .. tacv:: eapply @term
-      :name: eapply
+         apply H1 in H0.  (* change H0, create new goals for unmatched premises of H1 *)
 
-      The tactic :tacn:`eapply` behaves like :tacn:`apply` but it does not fail when no
-      instantiations are deducible for some variables in the premises. Rather, it
-      turns these variables into existential variables which are variables still to
-      instantiate (see :ref:`Existential-Variables`). The instantiation is
-      intended to be found later in the proof.
+   .. _apply_with_binding_goal:
+   .. example:: Apply a theorem with a binding in a goal
 
-   .. tacv:: rapply @term
-      :name: rapply
+      :tacn:`apply` unifies the conclusion `n <= p` of the theorem
+      `le_trans : forall n m p, n <= m -> m <= p -> n <= p`
+      with the goal, assigning `x * x ^ n` and `y * y ^ n` in the goal
+      to, repectively, `n` and `p` in theorem (backward reasoning).
+      The `with` clause provides the binding for `m`:
 
-      The tactic :tacn:`rapply` behaves like :tacn:`eapply` but it
-      uses the proof engine of :tacn:`refine` for dealing with
-      existential variables, holes, and conversion problems.  This may
+      .. coqtop:: reset none
+
+         Require Import PeanoNat.
+         Require Import Nat.
+         Infix "^" := pow : nat_scope.
+         Goal forall (x y n : nat), x * x ^ n <= y * y ^ n.
```

Comment:
I think that the `^ n` part complicates the example for no gain, I would remove it to keep only the multiplicative part. Also, I suggest adding `x <= y` in the hypothesis to make the goal credible.